src/routes/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import {
+  BrowserRouter as Router, Redirect, Route, Switch,
+} from 'react-router-dom';
+import { ROUTES } from '../utils/routes';
+import { getCurrentUser } from '../utils/auth';
+import LoginPage from '../pages/LoginPage';
+import SignUpPage from '../pages/SignUpPage';
+import HomePage from '../pages/HomePage';
+
+class AuthenticatedRoute extends Component {
+  constructor(props) {
+    super(props);
+    this.user = getCurrentUser();
+    this.isAuthenticated = this.user != null;
+  }
+
+  renderIfNotAuthenticated = component => (
+    this.props.path === ROUTES.login || this.props.path === ROUTES.register
+      ? (component)
+      : (<Redirect to={{ pathname: ROUTES.login }} />)
+  );
+
+  renderIfAuthenticated = component => (
+    this.props.path === ROUTES.login || this.props.path === ROUTES.register ? (
+      <Redirect
+        to={{ pathname: ROUTES.home, state: { from: this.props.location } }}
+      />
+    ) : (
+      component
+    )
+  );
+
+  render() {
+    const { component: Comp, ...rest } = this.props;
+    return (
+      <Route
+        {...rest}
```

Comment:
Add spaces between the brackets and the `rest` object.

Reply:
For this one, the eslint automatically formats back to no spaces.

Reply:
Okay I see, can you give it a more descriptive name,  I think rest is not descriptive enough

Reply:
All done. 👍